debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/ShrCCommand.java

```diff
@@ -112,28 +115,58 @@ public void run(String command, String[] args, Context context, PrintStream out)
 						String addr;
 						
 						addr = args[1].substring(0, args[1].indexOf(rangeDelim));
-						metaStart = U8Pointer.cast(CommandUtils.parsePointer(addr, J9BuildFlags.env_data64));
+						metaStart[0] = U8Pointer.cast(CommandUtils.parsePointer(addr, J9BuildFlags.env_data64));
 						
 						addr = args[1].substring(args[1].indexOf(rangeDelim) + rangeDelim.length());
-						metaEnd = U8Pointer.cast(CommandUtils.parsePointer(addr, J9BuildFlags.env_data64));
+						metaEnd[0] = U8Pointer.cast(CommandUtils.parsePointer(addr, J9BuildFlags.env_data64));
+						userSpecRange = true;
+					} else if (args[1].indexOf("layer=") != -1) {
+						int topLayer = dbgShrcCacheTopLayer(out, sharedClassConfig);
+						if (-1 == topLayer) {
+							CommandUtils.dbgPrint(out, "This cache does not have layers\n");
+							return;
+						}
+						Pattern p = Pattern.compile("(.*)layer=(\\d+)$");
+						Matcher m = p.matcher(args[1]);
+						if (m.matches()) {
+							String layerNumberString = m.group(2);
+							layer = Integer.parseInt(layerNumberString);
+							if (layer >= metaStartInCache.length) {
+								CommandUtils.dbgPrint(out, "User specified layer number %d is invalid, the maximum layer number is %d\n", layer, metaStartInCache.length - 1);
+								return;
+							}
+							metaStart[0] = U8Pointer.cast(metaStartInCache[layer]);
+							metaEnd[0] = U8Pointer.cast(metaEndInCache[layer]);
+							userSpecRange = true;
+						}
 					}
-				}
-				
-				if (metaStart != null && metaStart.longValue() < metaStartInCache.longValue()) {
-					CommandUtils.dbgPrint(out, "User specified start boundary is before metadata start in cache\n");
+					if (userSpecRange) {
+						if (metaEnd[0].lt(metaStart[0])) {
+							CommandUtils.dbgPrint(out, "User specified metadata region boundary is not valid. Ensure 'end' >= 'start'\n");
+							return;
+						}
+					
+						int index1 = getArrayIndexForMetadataAddress(metaStart[0], metaStartInCache, metaEndInCache);
+						int index2 = getArrayIndexForMetadataAddress(metaEnd[0], metaStartInCache, metaEndInCache);
+
+						if ((-1 == index1) && (-1 == index2)) {
+							CommandUtils.dbgPrint(out, "User specified metadata region boundary is not valid. Neither of the addresses is in the shared cache metadata region.\n");
+							return;
+						} else if ((-1 == index1) && (-1 != index2)) {
+							CommandUtils.dbgPrint(out, "User specified start boundary is not in the shared cache metadata region\n");
+							metaStart[0] = metaStartInCache[index2];
+						} else if ((-1 != index1) && (-1 == index2)) {
+							CommandUtils.dbgPrint(out, "User specified end boundary is not in the shared cache metadata region\n");
+							metaEnd[0] = metaEndInCache[index1];
+						} else if ((-1 != index1) && (-1 == index2)) {
```

Comment:
This condition is the same as the one above and so will only be tested when it is false.